source/common/upstream/health_checker_impl.cc

```diff
@@ -793,9 +793,12 @@ void GrpcHealthCheckerImpl::GrpcActiveHealthCheckSession::logHealthCheckStatus(
     case grpc::health::v1::HealthCheckResponse::UNKNOWN:
       service_status = "unknown";
       break;
+    case grpc::health::v1::HealthCheckResponse::SERVICE_UNKNOWN:
+      service_status = "service_unknown";
+      break;
     default:
       // Should not happen really, Protobuf should not parse undefined enums values.
```

Comment:
I think you can delete this comment. :)

Reply:
Please remove it. Protobuf does parse unknown enums unless we are validating it on top using PGV which we are not doing here.